tests/js/t8404.nim

```diff
@@ -0,0 +1,29 @@
+discard """
+  targets: "c cpp js"
+"""
+block: # bug #8404
+  # can conv
+  template float2int(T) =
+    var a = -1.0
+    let b = T(a)
+    doAssert b < 0
+    let c = b + 1
+    doAssert c is T
+    doAssert c == 0
+
+  float2int(int8)
+  float2int(int16)
+  float2int(int32)
+  float2int(int64)
+
```

Comment:
wrap everything inside a `template main()` and then:
```nim
main()
static: main()
```

Reply:
what's the purpose for?

Reply:
tests with nim vm; it's a common pattern

Reply:
you wrote:
```nim
static:
  main()
```

but forgot `main()` so it will only run in vm; as noted earlier use this:


```nim
main()
static: main()
```

so it runs both in vm and in RT